File: app/src/main/java/com/example/android/newsapp/refactor/mvp/contract/MainContract.java
package com.example.android.newsapp.refactor.mvp.contract;

import com.example.android.newsapp.refactor.model.Result;
import com.example.android.newsapp.refactor.mvp.base.BasePresenter;
import com.example.android.newsapp.refactor.mvp.base.BaseView;

import java.util.List;

public interface MainContract {

    interface View extends BaseView {
        void showNews(List<Result> news);
    }

    abstract class Presenter extends BasePresenter<View> {

    }
}

File: app/src/main/kotlin/com/example/android/newsapp/mvp/contract/MainContract.kt
package com.example.android.newsapp.mvp.contract

import com.example.android.newsapp.model.Result
import com.example.android.newsapp.mvp.base.BasePresenter
import com.example.android.newsapp.mvp.base.BaseView


interface MainContract {

    interface View : BaseView {
        fun showNews(news: List<Result>)
    }

    abstract class Presenter : BasePresenter<View>() {
        abstract fun getListSize(): Int
        abstract fun provideName(position: Int): String
    }
}
